### DISCUSSION TITLE
How to test react-native-tvos changes locally on Android TV

### DISCUSSION THREAD

**Discussion by danychi** [react-native-tvos/react-native-tvos#434] [Q&A] [answered]
Hi there,

I am busy creating a simple implementation to detect long presses on any DPad arrows by making changes right away on the `node_modules/react-native-tvos` in my current project. These changes seem to work right away on `tvOS` but not on `Android TV`. I have found this [article](https://reactnative.dev/contributing/how-to-build-from-source) from React Native where they explain how to build from source to test changes locally. After following all the steps, I'm getting an error that complains that `react-native can not be found`. I am currently using `react-native-tvos@0.69.5-1`.

So in synthesis, do you have any advice about how to run locally the changes made on `react-native-tvos` for Android TV?

**Comment by douglowder** [comment, marked as answer]
@danychi if the steps in that article don't work, you can follow these steps to generate the prebuilt libraries you need to copy into `node_modules/react-native` in your project.

You will need to have the right Android SDK and NDK versions installed.

- Clone this repo
- Checkout the `v0.69.5-1` tag since you are using that version of the repo
- In your copy of the repo, execute `yarn` to install dependencies
- Make your changes in the Android sources
- (Recommended) Change the version string to something local to your environment, e.g. `0.69.5-danychi`:

```bash
node scripts/set-rn-version -v 0.69.5-danychi
```

- Execute the script to build the Android libraries (hermes-engine and react-native):

```bash
node scripts/prepare-tv-release.js
# This will take a long time, especially the first time you do it, so go and get a coffee :)
```

This will create the `android` directory that goes in the release package. You should copy this directory into `node_modules/react-native` in your project -- then your Android builds there will pick up your changes.
